src/kibana/components/vislib/lib/dispatch.js

```diff
@@ -236,7 +237,7 @@ define(function (require) {
         .select('.legend-ul')
         .selectAll('li.color')
         .filter(function (d, i) {
-          return this.getAttribute('data-label') !== label;
+          return d.label.toString() !== label;
```

Comment:
It's generally safer to do `String(d.label)` or `(d.label + '')' rather than `d.label.toString()`